src/map.c

```diff
@@ -547,33 +547,31 @@ bool map_insert(map_t obj, void *key, void *value)
 
 static void map_prev(map_t obj, map_iter_t *it)
 {
-    if (!it->node) {
-        it->prev = NULL;
+    /* We have hit the end or null node. */
+    if (!it->node || it->node == obj->it_least.node) {
+        it->prev = it->node = NULL;
         return;
     }
 
-    if (it->node == obj->it_least.node) { /* We have hit the end. */
-        it->prev = it->node = NULL;
-    } else {
-        if (it->node->left) { /* To the left, as far right as possible */
-            it->node = it->node->left;
+    if (it->node->left) { /* To the left, as far right as possible */
+        it->node = it->node->left;
```

Comment:
Rewrite as following:
```c
    for (it->node = it->node->left; it->node->right; it->node = it->node->right)
        it->prev = it->node;
    return;
```

Later, to further simplify, we may create a `foreach` macro.

Reply:
Shall the foreach macro be included in this pull request?

Reply:
> Shall the foreach macro be included in this pull request?

Up to you. Since this pull request is intended to refactor the existing rbtree, we could think of feasible enhancements for both performance and simplicity.